src/librawspeed/decoders/NakedDecoder.cpp

```diff
@@ -44,50 +44,55 @@ NakedDecoder::NakedDecoder(FileMap* file, Camera* c) :
 
 NakedDecoder::~NakedDecoder() = default;
 
-RawImage NakedDecoder::decodeRawInternal() {
-  uint32 width=0, height=0, filesize=0, bits=0, offset=0;
-  if(cam->hints.find("full_width") != cam->hints.end()) {
-    string tmp = cam->hints.find(string("full_width"))->second;
-    width = (uint32) atoi(tmp.c_str());
-  } else
-    ThrowRDE("Naked: couldn't find width");
-
-  if(cam->hints.find("full_height") != cam->hints.end()) {
-    string tmp = cam->hints.find(string("full_height"))->second;
-    height = (uint32) atoi(tmp.c_str());
-  } else
-    ThrowRDE("Naked: couldn't find height");
-
-  if(cam->hints.find("filesize") != cam->hints.end()) {
-    string tmp = cam->hints.find(string("filesize"))->second;
-    filesize = (uint32) atoi(tmp.c_str());
-  } else
-    ThrowRDE("Naked: couldn't find filesize");
-
-  if(cam->hints.find("offset") != cam->hints.end()) {
-    string tmp = cam->hints.find(string("offset"))->second;
-    offset = (uint32) atoi(tmp.c_str());
-  }
+static const map<string, BitOrder> order2enum = {
+    {"plain", BitOrder_Plain},
+    {"jpeg", BitOrder_Jpeg},
+    {"jpeg16", BitOrder_Jpeg16},
+    {"jpeg32", BitOrder_Jpeg32},
+};
+
+void NakedDecoder::parseHints() {
```

Comment:
> right. but is this really an issue right now?

nope, just some final polishing after the issues were fixed. in rawspeed anyway.

> because the function call itself will add something to the total required stack size for the executing thread. 

yep, this is not as helpful *for the thread stack size*, only for limiting function size.

> (All that based on the somewhat limited knowledge I have on the topic, this has simply never been any issue in my whole developing career.)

take darktable-2.0.7 sources, alpine linux chroot, and try running it :)
you'll see stack overuse crashes, which are now gone in 2.2, after all these fixes, including rawspeed part.